### ISSUE TITLE
Products are not showing for new Customer Group

### ISSUE THREAD

**Issue by alpsws** [magento/magento2#38133]
### Preconditions and environment

Magento ver. 2.4.4 , Magento ver. 2.4.6
Create new customer group 
Assign new customer group to customer
Logged in on frontend 
And you are able to see filter but not the products.
If you try to access product directly you are able to see it but on category and search page products are not showing.

While debugging I get below query and found that there isn't any entry for customer group id 4 in catalog_product_index_price table. I truncated table and did indexing again but still no entry for new customer group id 4. 

SELECT `e`.*, `price_index`.`price`, `price_index`.`tax_class_id`, `price_index`.`final_price`, IF(price_index.tier_price IS NOT NULL, LEAST(price_index.min_price, price_index.tier_price), price_index.min_price) AS `minimal_price`, `price_index`.`min_price`, `price_index`.`max_price`, `price_index`.`tier_price`, IFNULL(review_summary.reviews_count, 0) AS `reviews_count`, IFNULL(review_summary.rating_summary, 0) AS `rating_summary`, `stock_status_index`.`stock_status` AS `is_salable` FROM `catalog_product_entity` AS `e` INNER JOIN `catalog_product_index_price` AS `price_index` ON price_index.entity_id = e.entity_id AND price_index.customer_group_id = '4' AND price_index.website_id = '1' LEFT JOIN `review_entity_summary` AS `review_summary` ON e.entity_id = review_summary.entity_pk_value AND review_summary.store_id = 1 AND review_summary.entity_type = (SELECT `review_entity`.`entity_id` FROM `review_entity` WHERE (entity_code = 'product')) INNER JOIN `cataloginventory_stock_status` AS `stock_status_index` ON e.entity_id = stock_status_index.product_id WHERE (stock_status_index.stock_status = 1) AND (e.entity_id IN (1028, 1015, 1002, 989, 976, 963, 950, 937, 924, 911, 898, 893)) ORDER BY FIELD(e.entity_id,1028,1015,1002,989,976,963,950,937,924,911,898,893)

Am I missing any configuration for new customer group?

Thanks,
Nayan

### Steps to reproduce

1. Create new customer group 
2. Assign new customer group to customer
3. Logged in on frontend 
4. And you are able to see filter but not the products.

### Expected result

Products should be display for new customer group also.

### Actual result

Layered navigation is showing but products are not showing.

![image](https://github.com/magento/magento2/assets/55439710/ad4fb010-6e89-4682-a92e-fc49c79812c1)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by ananth-iyer** [comment]
@alpsws Did you run re-index command? On PLP, products are fetched from ES.
Products might not be synced with Elastic search/Open search.

**Comment by engcom-Bravo** [comment]
@ananth-iyer Thanks for your contribution & collaboration over here.

Hi @alpsws,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.


**Steps to reproduce**

- Create new customer group
- Assign new customer group to customer
- Logged in on frontend
- And you are able to see filter but not the products.

<img width="1425" alt="Screenshot 2023-10-31 at 12 15 20 PM" src="https://github.com/magento/magento2/assets/51680745/c3ec555b-c59e-4486-a5b4-9312b1589743">


<img width="1223" alt="Screenshot 2023-10-31 at 12 15 37 PM" src="https://github.com/magento/magento2/assets/51680745/ea03cf97-692b-4c2b-bb42-4d61e8894df1">


<img width="1437" alt="Screenshot 2023-10-31 at 12 12 34 PM" src="https://github.com/magento/magento2/assets/51680745/491b1691-3287-49ef-9460-0a29a4ac05b1">



We have created new customer group and assigned to new customer and we are able to see and filter the products successfully.

Kindly recheck the behaviour on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.

Thanks.
